cloud/scope/client.go

```diff
@@ -7,19 +7,55 @@ import (
+type LinodeInstanceClient interface {
+	GetInstanceIPAddresses(ctx context.Context, linodeID int) (*linodego.InstanceIPAddressResponse, error)
+	ListInstances(ctx context.Context, opts *linodego.ListOptions) ([]linodego.Instance, error)
+	CreateInstance(ctx context.Context, opts linodego.InstanceCreateOptions) (*linodego.Instance, error)
+	BootInstance(ctx context.Context, linodeID int, configID int) error
+	ListInstanceConfigs(ctx context.Context, linodeID int, opts *linodego.ListOptions) ([]linodego.InstanceConfig, error)
+	GetInstanceDisk(ctx context.Context, linodeID int, diskID int) (*linodego.InstanceDisk, error)
+	ResizeInstanceDisk(ctx context.Context, linodeID int, diskID int, size int) error
+	CreateInstanceDisk(ctx context.Context, linodeID int, opts linodego.InstanceDiskCreateOptions) (*linodego.InstanceDisk, error)
+	GetInstance(ctx context.Context, linodeID int) (*linodego.Instance, error)
+	DeleteInstance(ctx context.Context, linodeID int) error
+}
+
+// LinodeVPCClient defines the methods that a Linode client must have to interact with Linode's VPC service.
+type LinodeVPCClient interface {
+	GetVPC(ctx context.Context, vpcID int) (*linodego.VPC, error)
+}
+
+// LinodeNodeBalancerClient defines the methods that a Linode client must have to interact with Linode's Node Balancer service.
+type LinodeNodeBalancerClient interface {
+	ListNodeBalancers(ctx context.Context, opts *linodego.ListOptions) ([]linodego.NodeBalancer, error)
+	CreateNodeBalancer(ctx context.Context, opts linodego.NodeBalancerCreateOptions) (*linodego.NodeBalancer, error)
+	CreateNodeBalancerConfig(ctx context.Context, nodebalancerID int, opts linodego.NodeBalancerConfigCreateOptions) (*linodego.NodeBalancerConfig, error)
+	DeleteNodeBalancerNode(ctx context.Context, nodebalancerID int, configID int, nodeID int) error
+	DeleteNodeBalancer(ctx context.Context, nodebalancerID int) error
+	CreateNodeBalancerNode(ctx context.Context, nodebalancerID int, configID int, opts linodego.NodeBalancerNodeCreateOptions) (*linodego.NodeBalancerNode, error)
+}
+
+// LinodeObjectStorageClient defines the methods that a Linode client must have to interact with Linode's Object Storage service.
 type LinodeObjectStorageClient interface {
 	GetObjectStorageBucket(ctx context.Context, cluster, label string) (*linodego.ObjectStorageBucket, error)
 	CreateObjectStorageBucket(ctx context.Context, opts linodego.ObjectStorageBucketCreateOptions) (*linodego.ObjectStorageBucket, error)
 	CreateObjectStorageKey(ctx context.Context, opts linodego.ObjectStorageKeyCreateOptions) (*linodego.ObjectStorageKey, error)
 	DeleteObjectStorageKey(ctx context.Context, keyID int) error
 }
 
-// LinodeObjectStorageClientBuilder is a function that returns a LinodeObjectStorageClient.
-type LinodeObjectStorageClientBuilder func(apiKey string) (LinodeObjectStorageClient, error)
+type LinodeObjectStorageBucketBuilder func(apiKey string) (LinodeObjectStorageClient, error)
```

Comment:
This function signature is a builder type that builds a client, so we should keep `ClientBuilder` as its suffix. I also intentionally used `ObjectStorage` and omitted bucket since it provisions buckets and keys as resources served by Linode's Object Storage API.

```suggestion
type LinodeObjectStorageClientBuilder func(apiKey string) (LinodeObjectStorageClient, error)
```